flytekit/remote/remote.py

```diff
@@ -229,7 +231,8 @@ def _get_pickled_target_dict(root_entity: typing.Union[WorkflowBase, PythonTask]
                 queue.append(task)
         elif isinstance(entity, CoreNode):
             queue.append(entity.flyte_entity)
-    return pickled_target_dict
+    md5_bytes = hashlib.md5(cloudpickle.dumps(pickled_target_dict)).digest()
```

Comment:
With this change, I think we end up calling `cloudpickle.dumps(pickled_dict)` twice, here and in `_pickle_and_upload_entity`. Is there a way to only pickle once?

Reply:
that's true. We can store the `pickled_target_dict` output at L1990, and send it into `_pickle_and_upload_entity()`. In `_pickle_and_upload_entity()`, it could compute pickle dict if `pickled_target_dict` is None